epoch.go

```diff
@@ -1641,11 +1717,20 @@ func (e *Epoch) processFinalizedBlock(block Block, finalization Finalization) er
 		return e.processReplicationState()
 	}
 
+	blockDependency, missingRounds := e.blockDependencies(block.BlockHeader())
+	// because its finalized we don't care about empty rounds
+	if blockDependency != nil {
+		e.Logger.Error(
```

Comment:
Can't we somehow call `processFinalizedBlock` on out of order blocks and reach here while we don't have the parent block?